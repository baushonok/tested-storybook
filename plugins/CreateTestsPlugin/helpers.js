const fs = require("fs");
const path = require("path");

const getComponentName = (fileContent, pattern) =>
  fileContent.match(pattern)[0];

const getComponentStoriesNames = (fileContent, pattern) =>
  fileContent.match(pattern);

const generateHermioneTest = (
  testDirectoryPath,
  componentName,
  componentStoryName,
  testTemplate
) => {
  const testPath = path.resolve(
    testDirectoryPath,
    `${componentStoryName}.hermione.js`
  );

  if (!fs.existsSync(testPath)) {
    fs.createWriteStream(testPath, "utf8");
    fs.writeFileSync(
      testPath,
      testTemplate(componentName, componentStoryName),
      "utf8"
    );
  }
};

module.exports = {
  getComponentName,
  getComponentStoriesNames,
  generateHermioneTest
};
