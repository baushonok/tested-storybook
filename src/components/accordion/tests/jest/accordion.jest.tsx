import { render, RenderResult } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";

import Accordion from "../..";
import { getContent, getToggle } from "./helpers";

describe("Accordion", () => {
  describe("shapshots", () => {
    test("should match when empty", () => {
      const tree = renderer.create(<Accordion />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test("should match with content", () => {
      const tree = renderer
        .create(<Accordion>Some content</Accordion>)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("behaviour", () => {
    let renderedComponent: RenderResult;
    let toggle: HTMLElement;

    beforeEach(() => {
      renderedComponent = render(<Accordion />);
      toggle = getToggle(renderedComponent)!;
    });

    test("should be collapsed by default", () => {
      const content = getContent(renderedComponent);

      expect(content).not.toBeInTheDocument();
    });

    test("should be expanded after clicking toggle", () => {
      toggle.click();

      const content = getContent(renderedComponent);

      expect(content).toBeInTheDocument();
    });

    test("should be collapsed by double after clicking toggle", () => {
      toggle.click();
      toggle.click();

      const content = getContent(renderedComponent);

      expect(content).not.toBeInTheDocument();
    });
  });
});
