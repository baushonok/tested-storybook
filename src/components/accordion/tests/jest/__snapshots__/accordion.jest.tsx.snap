// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion shapshots should match when empty 1`] = `
<button
  data-testid="accordion-toggle"
  onClick={[Function]}
  type="button"
>
  Click me
</button>
`;

exports[`Accordion shapshots should match with content 1`] = `
<button
  data-testid="accordion-toggle"
  onClick={[Function]}
  type="button"
>
  Click me
</button>
`;
